### PR TITLE
Firebase Integration

### PR DESCRIPTION
**Pull request by vivianliu0**
Items completed in the PR:
- Login page is integrated with firebase
- Sign up page creates a user and is saved in firebase
- Password Reset page, which sends a reset email if the one provided is valid.
- Added routing logic so admin page is only accessible by logged in user
- Integrated with Firestore database to store user information such as displayName

passwordReset

signup

admin

### CODE REVIEW

File: src/Components/Home/home.js
Comment (jaydhulia) [inline src/Components/Home/home.js:28]: Noice

**Comment by ahmedhamodi** [comment]
Thank you for putting up a WIP PR so we can see your current progress. :)

File: src/Components/Login/firebaseCredentials.js
Comment (ahmedhamodi) [inline src/Components/Login/firebaseCredentials.js:1]: Nice, this is a good idea. Thanks for doing this

File: src/Routes.js
Comment (jaydhulia) [inline src/Routes.js:12]: When a user is logged in, and they go to any of the pages ("/" or "/login" or "/signup"), it displays the original not signed in content first, then after a couple of seconds, it switches to Admin dashboard. Is that something that can be fixed?

**Comment by jaydhulia** [comment]
LGTM other than my 1 comment. Also the circleci is failing, so you may need to fix that. Nice work on this @vivianliu0!

File: src/Routes.js
Comment (vivianliu0) [reply src/Routes.js:12]: Hmm I think this is happening because the user object isn't fully loaded yet, I'm taking a look a possible ways we could fix this. One way would be to render a loader until the object is loaded

Comment (jaydhulia) [reply src/Routes.js:12]: Yup a loader is something I have used before and maybe handy for other parts of the project as well. @ahmedhamodi what are your thoughts on this? Don't want to be a blocker, so adding a `loader` type functionality can be an entirely different ticket, to add that type of functionality to the project in general.

Comment (ahmedhamodi) [reply src/Routes.js:12]: Yeah a loader/spinner is the way to go. But can be addressed later

Comment (jaydhulia) [reply src/Routes.js:12]: Cool, in that case, PR looks good to me!

File: src/firebaseCredentials.js
Comment (jaydhulia) [inline src/firebaseCredentials.js:33]: Should we be logging errors to the console here and on line 48?

Comment (vivianliu0) [reply src/firebaseCredentials.js:33]: oops, will remove this, I'm surprised the linters didn't pick this up

Comment (jaydhulia) [reply src/firebaseCredentials.js:33]: Yup seems like we are allowing it based on the eslint file (https://github.com/uwblueprint/cdc-frontend/blob/master/.eslintrc#L37). Is that expected behaviour @ahmedhamodi?

Comment (ahmedhamodi) [reply src/firebaseCredentials.js:33]: Yes console.errors are supposed to be allowed. Only log and info I think are caught by the linter, which is what I want. We can change that if the larger team disagrees though!

Comment (ahmedhamodi) [reply src/firebaseCredentials.js:33]: Oh, info is also allowed based on the current eslint file. I'd personally vote to remove that, but honestly no one is using console.info unless they have a reason for it, in which case it is probably fine to allow. Logs are used much more often and are the ones that might leak sensitive information to a client.

Comment (jaydhulia) [reply src/firebaseCredentials.js:33]: Cool, we can discuss that for later. For now, let's remove these errors since they are related to auth, just in case they contain anything sensitive :) @vivianliu0

Comment (vivianliu0) [reply src/firebaseCredentials.js]: Sounds good, removed

**Review by jaydhulia — DISMISSED** [review]

**Review by ahmedhamodi — CHANGES_REQUESTED** [review]
Let's not merge this just yet, I just synced with Jay offline and the reset password flow wasn't tested.

I'll test later tonight when I'm done with work and approve if everything looks fine. :)

File: src/Components/Login/signup.js
Comment (ahmedhamodi) [inline src/Components/Login/signup.js:139]: Should be allErrors.password

File: src/Components/Login/signup.js
Comment (ahmedhamodi) [inline src/Components/Login/signup.js:59]: Don't think this is needed from a signup perspective

**Review by ahmedhamodi — COMMENTED** [review]
Ran into a small bug, added comments that should address it

**Comment by ahmedhamodi** [comment]
Not sure if we have control over this code wise, but the project name isn't very pretty in the reset password email lol
image

**Comment by ahmedhamodi** [comment]
> Not sure if we have control over this code wise, but the project name isn't very pretty in the reset password email lol

Fixed this through the Firebase app settings, so don't worry about this.

File: src/Components/Login/signup.js
Comment (vivianliu0) [reply src/Components/Login/signup.js:139]: Thanks for catching this, after fixing it to allErrors.password it now provides a more helpful message when for example a weak password is given.

File: src/Components/Login/signup.js
Comment (vivianliu0) [reply src/Components/Login/signup.js]: Good point, removed

**Review by ahmedhamodi — APPROVED** [review]
LGTM! Thanks so much for doing this 🎉

**Pull request merged**
